nntrainer/tensor/tensor.cpp

```diff
@@ -219,8 +230,27 @@ int Tensor::add_i(float const &value) {
   return ML_ERROR_NONE;
 }
 
+Tensor Tensor::add(Tensor const &m, Tensor &out, float const alpha) const {
+  CREATE_IF_EMPTY_DIMS(out, getDim());
+  scopy(length(), getData(), 1, out.getData(), 1);
+  out.add_i(m, alpha);
+
+  return out;
+}
+
 Tensor Tensor::add(float const &value) { CLONE_OP_I(add_i, value); }
 
+Tensor Tensor::add(float const &value, Tensor &out) const {
+  const float *data = getData();
+
+  CREATE_IF_EMPTY_DIMS(out, getDim());
+
+  std::transform(data, data + length(), out.getData(),
+                 std::bind2nd(std::plus<float>(), value));
+
+  return out;
```

Comment:
so.. if we are using this... the dimensions is going to change. and if it is the same.. then.. we don't need to another buffer?

Reply:
you mean out dimension? if uninitialized, it is initialized with `this->getDim()`, if dimension is not correct it is UB 🍫 .

 🍫  is intended as the function is in critical path.

Reply:
On second thought... It generates a quite of bug which is really hard to find.

Reply:
I think it's good idea. I'll update on #870 since this patch is being depended upon other PR :)

Reply:
Update done :)